vault/data_source_auth_backends.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func authBackendsDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: ReadWrapper(authBackendsDataSourceRead),
+		Schema: map[string]*schema.Schema{
+			"paths": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The auth backend mount points.",
+			},
+			"type": {
```

Comment:
Can we use consts for each of these? If not already defined in https://github.com/hashicorp/terraform-provider-vault/blob/main/internal/consts/consts.go then we can add new entries.
```suggestion
			consts.FieldType: {
```